src/main.rs

```diff
@@ -1,3 +1,107 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io;
+
+#[derive(Deserialize)]
+struct SceneOption {
+    scene_id: String,
+    text: String,
+}
+#[derive(Deserialize)]
+struct Scene {
+    scene_id: String,
+    text: String,
+    end: bool,
+    win: bool,
+    options: Vec<SceneOption>,
+}
+
+impl Scene {
+    fn print_header(&self) {
+        let line: String = (0..self.text.len() + 2).map(|_| "-").collect();
+        println!("+{}+", line);
+        println!("| {} |", self.text);
+        println!("+{}+", line);
+    }
+
+    fn print_options(&self) {
+        for (index, option) in self.options.iter().enumerate() {
+            println!("  {}) {}", index + 1, option.text);
+        }
+    }
+}
+
+const FILE_PATH: &str = "./src/game.json";
+
+fn print_game_over() {
+    println!("Game over, here is a panda 🐼 to make you feel better.");
+    println!("If you wanna have some fun, see https://www.youtube.com/watch?v=dQw4w9WgXcQ");
+}
+
+fn print_game_win() {
+    println!("🍾🍾 You have won! 🍾🍾");
+    println!("https://www.youtube.com/watch?v=GC5E8ie2pdM")
+}
+
+fn get_scenes() -> HashMap<String, Scene> {
```

Comment:
jestli to dobre chapu, tak tady v podstate zafklam pamet 2x tim JSONem (jednou ve vektoru, podruhe v hashmape), nicmene na konci funkce se ten vektor z pameti uvolni?